### PR TITLE
URL encode names of new projects and events when passing in URL, and …

### PR DESCRIPTION
**Pull request by bhg**
…decode in alert header.

### CODE REVIEW

**Comment by bhg** [comment]
I decided in the end that passing project or event ID and fetching the title would involve an enormous amount of effort and obfuscation, and that simply encoding the URL correctly would be simpler, more intuitive and run faster.

I also noticed here that adding a new Group does not bring up an alert banner as do Projects and Events. Is this by design or should this be added?

**Review by marlonkeating — CHANGES_REQUESTED** [review]
This is a big improvement, but I saw at least one test case it didn't handle quite right

image

Repro query: projectAwaitingApproval=Test%20Project%20%24%26%2B%23%20/%5C-

I would recommend putting the encode/decode steps into helper methods that can be unit tested.

We do need to show this for the Create Group scenario, and I added an issue for that: https://github.com/DemocracyLab/CivicTechExchange/issues/680
If you'd like to wrap that fix into this PR you can.

**Comment by bhg** [comment]
> This is a big improvement, but I saw at least one test case it didn't handle quite right
> 
> image
> 
> Repro query: projectAwaitingApproval=Test%20Project%20%24%26%2B%23%20/%5C-
> 
Looks like this is happening because the replace() function is interpreting parts of the project name as a regex. There's actually no need to use replace() here - a simple concatenation would get rid of this weird behavior.

**Comment by bhg** [comment]
Implemented the above suggestions. The only thing missing right now is running the new tests to ensure that they passed.

File: common/components/chrome/AlertHeader.jsx [hunk 1]
```diff
@@ -178,10 +182,8 @@ class AlertHeader extends React.Component<Props, State> {
 
   _renderTriggeredAlert(): React$Node {
     const key: string = _.keys(AlertMessages).find(key => url.argument(key));
-    const message: string = AlertMessages[key].replace(
-      "{value}",
-      decodeURI(url.argument(key))
-    );
+    const message: string = AlertMessages[key] + '"' + url.decodeNameFromUrlPassing(url.argument(key)) + '"' 
+                          + AlertMessageEnding;
```
Comment (marlonkeating) [inline common/components/chrome/AlertHeader.jsx:186]: Avoiding duplicating code is generally good, but in this case I think the added code complexity to reduce duplication, coupled with difficulty future code editors might have in searching for these now chopped up front-end strings, would make me prefer the old approach.

Comment (bhg) [reply common/components/chrome/AlertHeader.jsx]: Agreed, and I did this to deal with Javascript's strange replace() functionality, which interprets sequences such as $& as flags.

In the end I realized that I can very easily void this by just performing the string replacement when the project/event/group name is still URL encoded.

**Review by marlonkeating — APPROVED** [review]

**Pull request merged**